### PR TITLE
floats: add round to precision function

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look @btracey.

### CODE REVIEW

File: floats.go [hunk 1]
```diff
@@ -594,6 +594,33 @@ func Prod(s []float64) float64 {
 	return prod
 }
 
+// Round returns the half away from zero rounded value of x with prec precision.
+//
+// Special cases are:
+// 	Round(±0) = +0
+// 	Round(±Inf) = ±Inf
+// 	Round(NaN) = NaN
+func Round(x float64, prec int) float64 {
```
Comment (btracey) [inline floats.go]: So to round to the 10s place one would use a negative precision?

Comment (kortschak) [inline floats.go]: Yes.

Comment (btracey) [inline floats.go]: Oops, sorry, I missed the two negative precision tests at the bottom.

**Comment by btracey** [comment]
LGTM.

How long has testing/quick existed? This is the first I've seen it.

**Comment by kortschak** [comment]
A very long time.

File: floats.go (hunk 1 above)
Comment (kortschak) [inline floats.go]: I added them because of your comment. Thanks.

**Pull request merged**